crates/bevy_winit/src/winit_windows.rs

```diff
@@ -25,59 +23,47 @@ impl WinitWindows {
     pub fn create_window(
         &mut self,
         event_loop: &winit::event_loop::EventLoopWindowTarget<()>,
-        window_id: WindowId,
-        window_descriptor: &WindowDescriptor,
-    ) -> Window {
+        entity: Entity,
+        window: &Window,
+    ) -> &winit::window::Window {
         let mut winit_window_builder = winit::window::WindowBuilder::new();
 
-        let &WindowDescriptor {
-            width,
-            height,
-            position,
-            monitor,
-            scale_factor_override,
-            ..
-        } = window_descriptor;
-
-        let logical_size = LogicalSize::new(width, height);
-
-        let monitor = match monitor {
-            MonitorSelection::Current => None,
-            MonitorSelection::Primary => event_loop.primary_monitor(),
-            MonitorSelection::Index(i) => event_loop.available_monitors().nth(i),
-        };
-
-        let selected_or_primary_monitor = monitor.clone().or_else(|| event_loop.primary_monitor());
-
-        winit_window_builder = match window_descriptor.mode {
-            WindowMode::BorderlessFullscreen => winit_window_builder
-                .with_fullscreen(Some(Fullscreen::Borderless(selected_or_primary_monitor))),
-            WindowMode::Fullscreen => winit_window_builder.with_fullscreen(Some(
-                Fullscreen::Exclusive(get_best_videomode(&selected_or_primary_monitor.unwrap())),
+        winit_window_builder = match window.mode {
+            WindowMode::BorderlessFullscreen => winit_window_builder.with_fullscreen(Some(
+                winit::window::Fullscreen::Borderless(event_loop.primary_monitor()),
             )),
             WindowMode::SizedFullscreen => winit_window_builder.with_fullscreen(Some(
-                Fullscreen::Exclusive(get_fitting_videomode(
-                    &selected_or_primary_monitor.unwrap(),
-                    window_descriptor.width as u32,
-                    window_descriptor.height as u32,
+                winit::window::Fullscreen::Exclusive(get_fitting_videomode(
+                    &event_loop.primary_monitor().unwrap(),
+                    window.width() as u32,
+                    window.height() as u32,
                 )),
             )),
-            WindowMode::Windowed => {
-                if let Some(sf) = scale_factor_override {
-                    winit_window_builder.with_inner_size(logical_size.to_physical::<f64>(sf))
-                } else {
-                    winit_window_builder.with_inner_size(logical_size)
+            _ => {
```

Comment:
Why the switch to the wildcard here? Do we really want to set the window position and size when in Fullscreen mode?

Reply:
I think this was what it was before https://github.com/bevyengine/bevy/pull/6934 and I prob just mis-merged that part of it.

Reply:
Ahhh yeah that makes sense!

Reply:
I was working on fixing this and I noticed another bug with the WindowMode stuff of monitor selection, so I'm probably going to make the full screen modes take a `MonitorSelection`.

Reply:
Actually hmm, maybe we should set the position/size even when fullscreening? Since we might switch to Windowed at some point depending on what the user wants.

Reply:
I'm going to defer this to a to a later PR, some of this isn't correct but it should be fine for now since it keeps generally the same behavior we had.

Reply:
Yeah I agree that we should defer that to a later PR.